NuRadioMC/SignalProp/radioproparaytracing.py

```diff
@@ -74,48 +71,34 @@ class describing the index-of-refraction profile
             loaded yaml config file
         detector: detector object
         """
-        self.__logger = logging.getLogger('ray_tracing')
-        self.__logger.setLevel(log_level)
+        super().__init__(medium=medium, 
+                         attenuation_model=attenuation_model,
+                         log_level=log_level,
+                         n_frequencies_integration=n_frequencies_integration, 
+                         n_reflections=n_reflections,
+                         config=config, 
+                         detector=detector)
 
         try:
             import radiopropa
         except ImportError:
-            self.__logger.error('ImportError: This raytracer depends on radiopropa which could not be imported. Check wether all dependencies are installed correctly. More information on https://github.com/nu-radio/RadioPropa')
+            self._logger.error('ImportError: This raytracer depends on radiopropa which could not be imported. Check wether all dependencies are installed correctly. More information on https://github.com/nu-radio/RadioPropa')
             raise ImportError('This raytracer depends on radiopropa which could not be imported. Check wether all dependencies are installed correctly. More information on https://github.com/nu-radio/RadioPropa')
-
-        self.__medium = medium
-        self.__ice_model = medium.get_ice_model_radiopropa()
-        self.__attenuation_model = attenuation_model
-        if(n_reflections):
-            if(not hasattr(self.__medium, "reflection") or self.__medium.reflection is None):
-                self.__logger.warning("ray paths with bottom reflections requested medium does not have any reflective layer, setting number of reflections to zero.")
-                n_reflections = 0
-        self.__n_reflections = n_reflections
-        self.__config = config
-        self.__n_frequencies_integration = n_frequencies_integration
-        self.__max_detector_frequency = None
-        self.__detector = detector
-        if self.__detector is not None:
-            for station_id in self.__detector.get_station_ids():
-                sampling_frequency = self.__detector.get_sampling_frequency(station_id, 0)
-                if self.__max_detector_frequency is None or sampling_frequency * .5 > self.__max_detector_frequency:
-                    self.__max_detector_frequency = sampling_frequency * .5
+        
+        self._ice_model = self._medium.get_ice_model_radiopropa()
 
         ## discard events if delta_C (angle off cherenkov cone) is too large
-        if config != None: 
-            self.__cut_viewing_angle = config['speedup']['delta_C_cut']*units.radian
+        if self._config != None: 
+            self._cut_viewing_angle = config['speedup']['delta_C_cut']*units.radian
         else: 
-            self.__cut_viewing_angle = 40*units.degree
+            self._cut_viewing_angle = 40*units.degree
```

Comment:
I guess now that the analytic ray tracer has a default config, we should implement this in a similar way for the other ray tracers for consistency?

Reply:
yes indeed, I saw that to and came to the same conclusion! :)